torchtune/datasets/_finetune.py

```diff
@@ -0,0 +1,136 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Any, Callable, Dict, Mapping, Optional
+
+import numpy as np
+
+from datasets import load_dataset
+from torch.utils.data import Dataset
+from torchtune.data import CROSS_ENTROPY_IGNORE_IDX, PromptTemplate
+from torchtune.modules.transforms import Transform
+
+
+class FinetuneDataset(Dataset):
+    """
+    Dataset class for creating instruct, chat, tool, or multimodal datasets for fine-tuning.
```

Comment:
No. This will replace both Instruct and ChatDataset. We are essentially merging the two, while also adding support for multimodal.

I haven't thought of a better name than FinetuneDataset. Maybe `TuneDataset` or `TokenizedDataset`?